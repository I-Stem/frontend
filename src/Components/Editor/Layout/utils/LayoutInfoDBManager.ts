import { getObject, putObject, saveObject } from "@utils/indexedDb";
import { Editor } from "../../EditorTypings";
import { BoundingBoxProperty } from "../CanvasTypes";
const dbName = "i-stem";

export type LayoutEditorBoundingBoxInfo = {
  [key in string]: BoundingBoxProperty[];
};

export const getLayoutEditorBoundingBoxForPage = async (
  fileId: string,
  currentPage: number
) => {
  return new Promise((resolve, reject) => {
    getObject(
      dbName,
      "layout",
      `${fileId}-box`,
      (data: LayoutEditorBoundingBoxInfo) => {
        if (!data || !data[currentPage]) {
          return resolve(null);
        }

        return resolve(data[currentPage]);
      }
    );
  });
};

/*
export const getLayoutAndTextInfoForPage = async (fileId: string, currentPage:number):Promise<{[key in string]:PageLayoutAndTextInfo} | null> => {
    return new Promise((resolve, reject) => {
        getObject("i-stem", "layout", `${fileId}-layout`, async (data: PagewiseLayoutAndTextInfo) => {
            if(data && currentPage in data)
            resolve(data[currentPage]);
            else
            resolve(null);
    });
});
}
*/

export const saveBoundingBoxesInIndexedDB = (
  fileId: string,
  currentPage: number,
  boundingBoxes: Array<BoundingBoxProperty>
) => {
  const keyPath = `${fileId}-box`;
  getObject(
    "i-stem",
    "layout",
    keyPath,
    (data: { [pageNumber: string]: BoundingBoxProperty[] }) => {
      if (data !== null) {
        putObject(
          dbName,
          "layout",
          { ...data, [currentPage]: boundingBoxes },
          "id",
          keyPath
        );
      } else
        saveObject(
          dbName,
          "layout",
          "id",
          { [currentPage]: boundingBoxes },
          keyPath
        );
    }
  );
};

export interface IndexedDBBoundingBoxSchema {
  //1 based page numbers
  [pageNumber: number]: BoundingBoxProperty[];
}
export const getBoundingBoxesForPageFromIndexedDb = async (
  fileId: string,
  currentPage: number
): Promise<BoundingBoxProperty[] | null> => {
  return new Promise((resolve, reject) => {
    getObject(
      dbName,
      "layout",
      `${fileId}-box`,
      (data: IndexedDBBoundingBoxSchema) => {
        if (data === null || !(currentPage in data)) {
          return resolve(null);
        }

        return resolve(data[currentPage]);
      }
    );
  });
};

export const updateCanvasBoxesWithReadingOrder = async (
  fileId: string,
  currentPage: number,
  boxes: BoundingBoxProperty[],
  layoutWithText: Editor.BoundingBoxWithCorrectedOCRText[]
) => {
  for (let i = 0; i < layoutWithText.length; i++) {
    boxes.filter(box => box.ind === layoutWithText[i].id)[0].order =
      layoutWithText[i].reading_order;
  }

  await saveBoundingBoxesInIndexedDB(fileId, currentPage, boxes);

  return boxes;
};

export const updateExistingBoundingBoxes = async (
  fileId: string,
  currentPage: number,
  updatedBoxes: BoundingBoxProperty[]
) => {
  const boxes =
    (await getBoundingBoxesForPageFromIndexedDb(fileId, currentPage)) || [];
  const result = updatedBoxes.forEach(box => {
    const boxIndex = boxes.findIndex(oldBox => box.ind === oldBox.ind);
    boxes[boxIndex] = box;
  });

  saveBoundingBoxesInIndexedDB(fileId, currentPage, boxes);
};
