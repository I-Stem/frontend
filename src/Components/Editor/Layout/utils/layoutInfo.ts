import { Editor, State } from "@Components/Editor/EditorTypings";
import { PageActivityInterface } from "@Interfaces";
import { EditorServices } from "@Services/API/Editor";
import { getObject, putObject, saveObject } from "@utils/indexedDb";
import { RefObject } from "react";
import { BoundingBoxProperty } from "../CanvasTypes";
import { toJson } from "./canvasUtils";
import { mapCanvasBoundingBoxesToRemediationAPISchema } from "./parseData";

/*
export interface PageLayoutAndTextInfo {
  layoutInfo: Editor.BoundingBoxWithCorrectedOCRText;
}

export interface PagewiseLayoutAndTextInfo {
  [key: string]: {
    [key in State]: PageLayoutAndTextInfo;
  };
}

const versionMaintainerForLayoutInfo = (
  data: PagewiseLayoutAndTextInfo,
  currentPage: number,
  layoutInfo: Editor.BoundingBoxWithCorrectedOCRText[]
) => {
  let pageData = {};
  if (currentPage in data) {
    const currentPageData = data[currentPage];
    if (1 in currentPageData) {
      //delete currentPageData[0];
      pageData = {
        0: { layoutInfo: currentPageData[1].layoutInfo },
        1: { layoutInfo },
      };
    } else {
      pageData = {
        ...currentPageData,
        1: { layoutInfo },
      };
    }
  } else {
    pageData = {
      0: {
        layoutInfo,
      },
    };
  }

  return pageData;
};
*/

export const getRelativeBoxesInX1Y1X2Y2FromRelativeXYWH = (
  relativeXYWH: Editor.BoundingBoxType
): Editor.BoundingBoxType => {
  return [
    relativeXYWH[0],
    relativeXYWH[1],
    relativeXYWH[0] + relativeXYWH[2],
    relativeXYWH[1] + relativeXYWH[3],
  ];
};

export const getRelativeBoxInXYWHFromX1Y1X2Y2 = (
  relativeX1Y1X2Y2: Editor.BoundingBoxType
): Editor.BoundingBoxType => {
  return [
    relativeX1Y1X2Y2[0],
    relativeX1Y1X2Y2[1],
    relativeX1Y1X2Y2[2] - relativeX1Y1X2Y2[0],
    relativeX1Y1X2Y2[3] - relativeX1Y1X2Y2[1],
  ];
};

export const getScaledUpBox = (
  relativeBox: Editor.BoundingBoxType,
  canvasWidth: number,
  canvasHeight: number
): Editor.BoundingBoxType => {
  return [
    relativeBox[0] * canvasWidth,
    relativeBox[1] * canvasHeight,
    relativeBox[2] * canvasWidth,
    relativeBox[3] * canvasHeight,
  ];
};

export const getScaledDownBox = (
  absoluteBox: Editor.BoundingBoxType,
  canvasWidth: number,
  canvasHeight: number
) => {
  return [
    absoluteBox[0] / canvasWidth,
    absoluteBox[1] / canvasHeight,
    absoluteBox[2] / canvasWidth,
    absoluteBox[3] / canvasHeight,
  ];
};

export const getTextInfoForLayoutFromRemediationAPI = async (
  fileId: string,
  currentPage: number,
  boundingBoxes: Array<BoundingBoxProperty>,
  twoColumn: boolean,
  analytics: PageActivityInterface,
  type: string,
  version: string
): Promise<{
  hasError: boolean;
  errorMessage: string | null;
  data: Editor.BoundingBoxWithCorrectedOCRText[];
}> =>
  new Promise((resolve, reject) => {
    toJson(`${fileId}-json`, async (json: Editor.OCRJsonSchema) => {
      const layoutInfo = mapCanvasBoundingBoxesToRemediationAPISchema(
        boundingBoxes,
        currentPage - 1,
        json,
        type,
        version,
        twoColumn
      );
      if (!layoutInfo) {
        return resolve({
          hasError: true,
          errorMessage:
            "Couldn't make request for text, error while data conversion",
          data: [],
        });
      }

      try {
        const res = await EditorServices.layoutText(currentPage, fileId, {
          analytics,
          recreationApi: layoutInfo,
        });

        console.log("api response: " + JSON.stringify(res));
        if (!res) {
          return resolve({
            hasError: true,
            errorMessage:
              "Couldn't get text data from backend, please try again.",
            data: [],
          });
        }
        if (res.code !== 200) {
          return resolve({
            hasError: true,
            errorMessage: res.message,
            data: [],
          });
        } else {
          return resolve({
            hasError: false,
            errorMessage: null,
            data: res.layoutInfo,
          });
        }
      } catch (error) {
        return resolve({
          hasError: true,
          errorMessage: "Couldn't call text API.",
          data: [],
        });
      }
    });
    console.log("exiting backend api call");
  });
