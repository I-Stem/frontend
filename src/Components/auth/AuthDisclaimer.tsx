import React from "react";
import Link from "next/link";
import { IAuth } from "./Auth";

function AuthDisclaimer(props: IAuth.IAuthDisclaimerProps) {
  const { message } = props;
  return (
    <p className="text-center text-gray-500">
      By clicking “{message}” , you agree to our{" "}
      <a href="https://www.istemai.com/istem_privacy_policy.html">
        <span className="disclaimer">Terms of Service and Privacy Policy.</span>
      </a>
    </p>
  );
}

export default AuthDisclaimer;
