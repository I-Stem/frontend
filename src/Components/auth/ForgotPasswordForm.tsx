import React, { useEffect, useRef } from "react";
import Head from "next/head";
import getConfig from "next/config";
import { connect } from "react-redux";
import { Button, Form, Input, Typography } from "antd";
import { AuthActions } from "@Actions";
import { IStore } from "@Interfaces";
import "./auth.scss";
import { IAuth } from "./Auth";

const { Title } = Typography;
const {
  publicRuntimeConfig: { BASE_URL, RESET_PASSWORD_URL },
} = getConfig();

const layout = {
  labelCol: { span: 24 },
  wrapperCol: { span: 24 },
};

const ForgotPasswordForm = (props: IAuth.IForgotPasswordProps) => {
  const { message } = props;
  const headerRef = useRef<HTMLDivElement>(null);
  useEffect(() => {
    headerRef.current?.focus();
    return () => {
      // Clear message state on component unmount
      props.clearAuthMessage();
    };
  }, []);

  const onFinish = (values: any) => {
    const { email } = values;
    const resetPasswordURL = `${BASE_URL}${RESET_PASSWORD_URL}`;
    props.forgot({ email, resetPasswordURL });
  };

  return (
    <div className="mt-16  auth-form">
      <Head>
        <title>Forgot password? | I-Stem</title>
      </Head>
      <Title className="lipHead">
        <div ref={headerRef} tabIndex={-1}>
          Forgot password?
        </div>
      </Title>
      <Title className="lipHead" level={4}>
        Reset your password
      </Title>
      <div className="h-8" />
      <Form
        layout="horizontal"
        name="basic"
        initialValues={{ remember: true }}
        onFinish={onFinish}
        {...layout}
        hideRequiredMark
        aria-live="polite"
      >
        <Form.Item
          label="Email Address you registered with"
          name="email"
          rules={[
            { required: true, message: "Please input your email!" },
            {
              type: "email",
              message: "Not a valid email address",
            },
          ]}
        >
          <Input size="large" placeholder="user@email.com" aria-live="off" />
        </Form.Item>
        <div className="warning-msg text-red-400">{message?.toString()}</div>
        <Form.Item className="mt-4">
          <Button
            size="large"
            className="login-button mt-4"
            block
            type="primary"
            htmlType="submit"
          >
            SEND PASSWORD RESET EMAIL
          </Button>
        </Form.Item>
      </Form>
    </div>
  );
};
function mapStateToProps(store: IStore) {
  const { message } = store;

  return {
    message: message?.message,
  };
}
const mapDispatchToProps = {
  forgot: AuthActions.Forgot,
  clearAuthMessage: AuthActions.ClearAuthMessage,
};
export default connect(mapStateToProps, mapDispatchToProps)(ForgotPasswordForm);
