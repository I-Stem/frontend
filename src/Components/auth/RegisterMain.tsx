import React from "react";
import Head from "next/head";
import { Button, Typography } from "antd";
import Link from "next/link";

import "./auth.scss";
import {
  REGISTER_BUSINESS_ROUTE,
  REGISTER_STUDENT_ROUTE,
  REGISTER_VOLUNTEER_ROUTE,
} from "@Definitions/Constants";
import volunteer from "@Pages/register/volunteer";
import { UserType } from "@Definitions/Constants";

const { Title } = Typography;

interface RegisterMainProps {
  registrationContext: string | string[];
}

const RegisterMain = (props: RegisterMainProps) => {
  return (
    <section id="registermain" className="mt-16 auth-form">
      <Head>
        <title>Register | I-Stem</title>
      </Head>
      <Title className="lipHead">Welcome to I-Stem!</Title>
      <Title className="lipHead" level={4}>
        Register your account as
      </Title>
      <div className="mt-6">
        <Link
          href={{
            pathname: REGISTER_STUDENT_ROUTE,
            query: {
              userType: UserType.I_STEM,
            },
          }}
        >
          <Button className="register-buttons mt-6" block>
            INDIVIDUAL
          </Button>
        </Link>
        <Link
          href={{
            pathname: REGISTER_BUSINESS_ROUTE,
            query: { userType: UserType.UNIVERSITY },
          }}
        >
          <Button className="register-buttons mt-6" block>
            UNIVERSITY REPRESENTATIVE
          </Button>
        </Link>
        <Link
          href={{
            pathname: REGISTER_BUSINESS_ROUTE,
            query: { userType: UserType.BUSINESS },
          }}
        >
          <Button className="register-buttons mt-6" block>
            BUSINESS REPRESENTATIVE
          </Button>
        </Link>
        <Link
          href={{
            pathname: REGISTER_STUDENT_ROUTE,
            query: { userType: UserType.VOLUNTEER },
          }}
        >
          <Button className="register-buttons mt-6" block>
            VOLUNTEER OR MENTOR
          </Button>
        </Link>
      </div>
    </section>
  );
};

export default RegisterMain;
