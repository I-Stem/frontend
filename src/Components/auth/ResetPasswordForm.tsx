import React, { useEffect, useRef } from "react";
import Head from 'next/head';
import { connect } from "react-redux";
import { useRouter } from "next/router";
import { Button, Form, Input, Typography } from "antd";
import { AuthActions } from "@Actions";
import "./auth.scss";
import { IAuth } from "./Auth";

const { Title, Text } = Typography;

const layout = {
  labelCol: { span: 24 },
  wrapperCol: { span: 24 },
};

const ResetPasswordForm = (props: IAuth.IResetPasswordProps) => {
  const router = useRouter();
  const headRef = useRef<HTMLDivElement>(null);
  useEffect(() => {
    headRef.current?.focus();
  }, []);
  const onFinish = (values: any) => {
    const { password } = values;
    const { email, resetToken } = router.query;
    props
      .resetPassword({
        password,
        passwordResetToken: resetToken.toString(),
        email: email.toString(),
      })
      .then(() => {
        router.push("/login");
      });
  };

  return (
    <div className="mt-16  auth-form">
      <Head>
        <title>Reset password | I-Stem</title>
      </Head>
      <Title className="lipHead">
        <div ref={headRef} tabIndex={-1}>
          Set a new password
        </div>
      </Title>
      <Text className="lip-head">
        Keep a strong password to protect your account, We recommend using a mix
        of upper case, lower case, numbers and special characters.
      </Text>
      <div className="h-8" />
      <Form
        layout="horizontal"
        name="basic"
        initialValues={{ remember: true }}
        onFinish={onFinish}
        {...layout}
        hideRequiredMark
        aria-live="polite"
      >
        <Form.Item
          {...layout}
          label="New Password"
          name="password"
          aria-live="polite"
          rules={[{ required: true, message: "Please enter a password" }]}
        >
          <Input.Password
            aria-live="off"
            placeholder="8+ characters"
            size="large"
          />
        </Form.Item>

        <Form.Item
          {...layout}
          aria-live="polite"
          label="Confirm New Password"
          name="password2"
          dependencies={["password"]}
          rules={[
            { required: true, message: "Please confirm your password" },
            ({ getFieldValue }) => ({
              validator(rule, value) {
                if (!value || getFieldValue("password") === value) {
                  return Promise.resolve();
                }
                return Promise.reject(
                  "The two passwords that you entered do not match!"
                );
              },
            }),
          ]}
          validateTrigger="onSubmit"
        >
          <Input.Password
            aria-live="off"
            placeholder="8+ characters"
            size="large"
          />
        </Form.Item>
        <Form.Item>
          <Button
            size="large"
            className="login-button mt-4 "
            block
            type="primary"
            htmlType="submit"
          >
            RESET PASSWORD
          </Button>
        </Form.Item>
      </Form>
    </div>
  );
};
const mapDispatchToProps = {
  resetPassword: AuthActions.ResetPassword,
};
export default connect(null, mapDispatchToProps)(ResetPasswordForm);
