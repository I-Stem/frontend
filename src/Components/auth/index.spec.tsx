import * as React from "react";

import { render } from "@Test/utils";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
// import { Rec } from "@Components";
import AuthDisclaimer from "./AuthDisclaimer";
import ForgotPasswordForm from "./ForgotPasswordForm";
import RegisterMain from "./RegisterMain";
import ResetPasswordForm from "./ResetPasswordForm";

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const store = mockStore({
  auth: {
    user: {},
  },
});
const baseProps = {
  toggleDialog: jest.fn(),
};
const ForgotPasswordFormStore = () => (
  <Provider store={store}>
    <ForgotPasswordForm />
  </Provider>
);

const RegisterMainStore = () => (
  <Provider store={store}>
    <RegisterMain registrationContext={[]} />
  </Provider>
);

const ResetPasswordFormStore = () => (
  <Provider store={store}>
    <ResetPasswordForm />
  </Provider>
);

describe("AuthDisclaimer", () => {
  it("should render without fail", () => {
    const { getByText } = render(<AuthDisclaimer message="Sign in" />);

    expect(getByText("Terms of Service and Privacy Policy.")).toBeTruthy();
  });

  it("should match snapshot", () => {
    const { container } = render(<AuthDisclaimer message="Sign in" />);

    expect(container).toMatchSnapshot();
  });
});
describe("ForgotPasswordForm", () => {
  it("should render without fail", () => {
    const { getByText } = render(<ForgotPasswordFormStore />);

    expect(getByText("Forgot password?")).toBeTruthy();
  });

  it("should match snapshot", () => {
    const { container } = render(<ForgotPasswordFormStore />);

    expect(container).toMatchSnapshot();
  });
});
describe("RegisterMain", () => {
  it("should render without fail", () => {
    const { getByText } = render(<RegisterMainStore />);

    expect(getByText("Register your account as")).toBeTruthy();
  });

  it("should match snapshot", () => {
    const { container } = render(<RegisterMainStore />);

    expect(container).toMatchSnapshot();
  });
});
describe("ResetPassword", () => {
  it("should render without fail", () => {
    const { getByText } = render(<ResetPasswordFormStore />);

    expect(getByText("Set a new password")).toBeTruthy();
  });

  it("should match snapshot", () => {
    const { container } = render(<ResetPasswordFormStore />);

    expect(container).toMatchSnapshot();
  });
});
