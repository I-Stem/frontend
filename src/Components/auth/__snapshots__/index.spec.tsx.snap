// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthDisclaimer should match snapshot 1`] = `
<div>
  <p
    class="text-center text-gray-500"
  >
    By clicking “
    Sign in
    ” , you agree to our
     
    <a
      href="https://www.istemai.com/istem_privacy_policy.html"
    >
      <span
        class="disclaimer"
      >
        Terms of Service and Privacy Policy.
      </span>
    </a>
  </p>
</div>
`;

exports[`ForgotPasswordForm should match snapshot 1`] = `
<div>
  <div
    class="mt-16  auth-form"
  >
    <h1
      class="ant-typography lipHead"
    >
      <div
        tabindex="-1"
      >
        Forgot password?
      </div>
    </h1>
    <h4
      class="ant-typography lipHead"
    >
      Reset your password
    </h4>
    <div
      class="h-8"
    />
    <form
      aria-live="polite"
      class="ant-form ant-form-horizontal ant-form-hide-required-mark"
      id="basic"
    >
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-col-24 ant-form-item-label"
        >
          <label
            class="ant-form-item-required"
            for="basic_email"
            title="Email Address you registered with"
          >
            Email Address you registered with
          </label>
        </div>
        <div
          class="ant-col ant-col-24 ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                aria-live="off"
                class="ant-input ant-input-lg"
                id="basic_email"
                placeholder="user@email.com"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="warning-msg text-red-400"
      />
      <div
        class="ant-row ant-form-item mt-4"
      >
        <div
          class="ant-col ant-col-24 ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <button
                class="ant-btn ant-btn-primary ant-btn-lg ant-btn-block login-button mt-4"
                type="submit"
              >
                <span>
                  SEND PASSWORD RESET EMAIL
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`RegisterMain should match snapshot 1`] = `
<div>
  <section
    class="mt-16 auth-form"
    id="registermain"
  >
    <h1
      class="ant-typography lipHead"
    >
      Welcome to I-Stem!
    </h1>
    <h4
      class="ant-typography lipHead"
    >
      Register your account as
    </h4>
    <div
      class="mt-6"
    >
      <button
        class="ant-btn ant-btn-block register-buttons mt-6"
        type="button"
      >
        <span>
          INDIVIDUAL
        </span>
      </button>
      <button
        class="ant-btn ant-btn-block register-buttons mt-6"
        type="button"
      >
        <span>
          UNIVERSITY REPRESENTATIVE
        </span>
      </button>
      <button
        class="ant-btn ant-btn-block register-buttons mt-6"
        type="button"
      >
        <span>
          BUSINESS REPRESENTATIVE
        </span>
      </button>
      <button
        class="ant-btn ant-btn-block register-buttons mt-6"
        type="button"
      >
        <span>
          VOLUNTEER OR MENTOR
        </span>
      </button>
    </div>
  </section>
</div>
`;

exports[`ResetPassword should match snapshot 1`] = `
<div>
  <div
    class="mt-16  auth-form"
  >
    <h1
      class="ant-typography lipHead"
    >
      <div
        tabindex="-1"
      >
        Set a new password
      </div>
    </h1>
    <span
      class="ant-typography lip-head"
    >
      Keep a strong password to protect your account, We recommend using a mix of upper case, lower case, numbers and special characters.
    </span>
    <div
      class="h-8"
    />
    <form
      aria-live="polite"
      class="ant-form ant-form-horizontal ant-form-hide-required-mark"
      id="basic"
    >
      <div
        aria-live="polite"
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-col-24 ant-form-item-label"
        >
          <label
            class="ant-form-item-required"
            for="basic_password"
            title="New Password"
          >
            New Password
          </label>
        </div>
        <div
          class="ant-col ant-col-24 ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <span
                class="ant-input-affix-wrapper ant-input-affix-wrapper-lg ant-input-password ant-input-password-large"
              >
                <input
                  action="click"
                  aria-live="off"
                  class="ant-input ant-input-lg"
                  id="basic_password"
                  placeholder="8+ characters"
                  type="password"
                  value=""
                />
                <span
                  class="ant-input-suffix"
                >
                  <span
                    aria-label="eye-invisible"
                    class="anticon anticon-eye-invisible ant-input-password-icon"
                    role="img"
                    tabindex="-1"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="eye-invisible"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                      />
                      <path
                        d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                      />
                    </svg>
                  </span>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-live="polite"
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-col-24 ant-form-item-label"
        >
          <label
            class="ant-form-item-required"
            for="basic_password2"
            title="Confirm New Password"
          >
            Confirm New Password
          </label>
        </div>
        <div
          class="ant-col ant-col-24 ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <span
                class="ant-input-affix-wrapper ant-input-affix-wrapper-lg ant-input-password ant-input-password-large"
              >
                <input
                  action="click"
                  aria-live="off"
                  class="ant-input ant-input-lg"
                  id="basic_password2"
                  placeholder="8+ characters"
                  type="password"
                  value=""
                />
                <span
                  class="ant-input-suffix"
                >
                  <span
                    aria-label="eye-invisible"
                    class="anticon anticon-eye-invisible ant-input-password-icon"
                    role="img"
                    tabindex="-1"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="eye-invisible"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                      />
                      <path
                        d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                      />
                    </svg>
                  </span>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-col-24 ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <button
                class="ant-btn ant-btn-primary ant-btn-lg ant-btn-block login-button mt-4 "
                type="submit"
              >
                <span>
                  RESET PASSWORD
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
